### PR TITLE
Version bump to V3

### PR DESCRIPTION
**Pull request by jayco**
There was a breaking change in https://github.com/buildkite/go-buildkite/pull/89 so I have done the things needed for a major version bump.

- [x] update import paths
- [x] update version
- [x] update changelog
- [x] go version minor bump
- [x] go mod tidy

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
-module github.com/buildkite/go-buildkite/v2
+module github.com/buildkite/go-buildkite/v3
 
-go 1.15
+go 1.16
```
Comment (keithduncan) [inline go.mod:3]: This bump [drops support for macOS 10.11](https://github.com/buildkite/agent/blob/main/docs/macos.md) which is likely fine I just wanted to note it 😄 we don’t support 10.11 in the agent any more either

**Review by keithduncan — APPROVED** [review]
Looks good to me, 100% your call on whether to do a major bump vs a backwards compatible change

File: go.mod (hunk 1 above)
Comment (jayco) [reply go.mod:3]: Good to know!

**Review by jradtilbrook — APPROVED** [review]
Ohh exciting! LGTM 👍

**Pull request merged**